# Réalisez une étude santé publique

In [1]:
## Question 1

In [2]:
import pandas as pd

In [3]:
import numpy as np

In [4]:
df1 = []

In [5]:
df1 = pd.read_csv("Q1.csv", usecols = ["Pays", "Valeur"])

In [6]:
df1

Pays   Valeur
0                               Afghanistan    30552
1                            Afrique du Sud    52776
2                                   Albanie     3173
3                                   Algérie    39208
4                                 Allemagne    82727
5                                    Angola    21472
6                        Antigua-et-Barbuda       90
7                           Arabie saoudite    28829
8                                 Argentine    41446
9                                   Arménie     2977
10                                Australie    23343
11                                 Autriche     8495
12                              Azerbaïdjan     9413
13                                  Bahamas      377
14                               Bangladesh   156595
15                                  Barbade      285
16                                  Bélarus     9357
17                                 Belgique    11104
18                                   Belize      332
19                                    Bénin    10323
20                                 Bermudes       65
21          Bolivie (État plurinational de)    10671
22                       Bosnie-Herzégovine     3829
23                                 Botswana     2021
24                                   Brésil   200362
25                        Brunéi Darussalam      418
26                                 Bulgarie     7223
27                             Burkina Faso    16935
28                               Cabo Verde      499
29                                 Cambodge    15135
..                                      ...      ...
145                                 Sénégal    14133
146                                  Serbie     9511
147                            Sierra Leone     6092
148                               Slovaquie     5450
149                                Slovénie     2072
150                                  Soudan    37964
151                               Sri Lanka    21273
152                                   Suède     9571
153                                  Suisse     8078
154                                Suriname      539
155                                Eswatini     1250
156                             Tadjikistan     8208
157                                   Tchad    12825
158                           Tchéquie (la)    10702
159                               Thaïlande    67011
160                             Timor-Leste     1133
161                                    Togo     6817
162                       Trinité-et-Tobago     1341
163                                 Tunisie    10997
164                            Turkménistan     5240
165                                 Turquie    74933
166                                 Ukraine    45239
167                                 Uruguay     3407
168                                 Vanuatu      253
169  Venezuela (République bolivarienne du)    30405
170                                Viet Nam    91680
171                                   Yémen    24407
172                                  Zambie    14539
173                                Zimbabwe    14150
174                                   Chine  1416667

[175 rows x 2 columns]

In [7]:
df1['Valeur'].sum() * 1000

8413993000

- Ce chiffre est supérieur à la population mondiale officielle. Il englobe deux fois la chine continentale.

In [8]:
df1_wo_china = df1[df1['Pays'] != "Chine"]

In [9]:
df1_wo_china['Valeur'].sum() * 1000

6997326000

- C'est une valeur plus conforme aux chiffres officiels (Source: https://www.prb.org/wp-content/uploads/2014/02/2013-population-data-sheet_french.pdf)

In [10]:
temp = pd.read_csv("pop.csv")

In [11]:
temp.columns = ['xx','xx2','country_code','country','xx3','element'
    ,'item_code','item','xx4',"year","unit","value",'xx5','xx6']

In [12]:
pop = temp.pivot_table(
    index=["country_code","country","item_code","item","year"],
    columns = ["element"], values=["value"], aggfunc=sum)

In [13]:
pop.columns = ['total_population']

In [14]:
pop.reset_index()
pop.query('country == "France"')

total_population
country_code country item_code item       year                  
68           France  2501      Population 2013             64291

## Question 2

In [15]:
df2 = pd.read_csv("Q2.csv")

In [16]:
df2

Code Domaine              Domaine  Code Pays    Pays  Code Élément  \
0           FBS  Bilans Alimentaires         68  France          5511   
1           FBS  Bilans Alimentaires         68  France          5611   
2           FBS  Bilans Alimentaires         68  France          5072   
3           FBS  Bilans Alimentaires         68  France          5911   
4           FBS  Bilans Alimentaires         68  France          5301   
5           FBS  Bilans Alimentaires         68  France          5521   
6           FBS  Bilans Alimentaires         68  France          5527   
7           FBS  Bilans Alimentaires         68  France          5123   
8           FBS  Bilans Alimentaires         68  France          5131   
9           FBS  Bilans Alimentaires         68  France          5154   
10          FBS  Bilans Alimentaires         68  France          5142   
11          FBS  Bilans Alimentaires         68  France           645   
12          FBS  Bilans Alimentaires         68  France           664   
13          FBS  Bilans Alimentaires         68  France           674   
14          FBS  Bilans Alimentaires         68  France           684   

                                              Élément  Code Produit Produit  \
0                                          Production          2511     Blé   
1                             Importations - Quantité          2511     Blé   
2                                  Variation de stock          2511     Blé   
3                             Exportations - Quantité          2511     Blé   
4                            Disponibilité intérieure          2511     Blé   
5                               Aliments pour animaux          2511     Blé   
6                                            Semences          2511     Blé   
7                                              Pertes          2511     Blé   
8                                          Traitement          2511     Blé   
9                                 Autres Utilisations          2511     Blé   
10                                         Nourriture          2511     Blé   
11  Disponibilité alimentaire en quantité (kg/pers...          2511     Blé   
12     Disponibilité alimentaire (Kcal/personne/jour)          2511     Blé   
13  Disponibilité de protéines en quantité (g/pers...          2511     Blé   
14  Disponibilité de matière grasse en quantité (g...          2511     Blé   

    Code Année  Année               Unité    Valeur Symbole  \
0         2013   2013  Milliers de tonnes  38614.00       S   
1         2013   2013  Milliers de tonnes   2055.00       S   
2         2013   2013  Milliers de tonnes   1131.00       S   
3         2013   2013  Milliers de tonnes  21502.00       S   
4         2013   2013  Milliers de tonnes  20298.00       S   
5         2013   2013  Milliers de tonnes   7822.00       S   
6         2013   2013  Milliers de tonnes    748.00       S   
7         2013   2013  Milliers de tonnes    358.00       S   
8         2013   2013  Milliers de tonnes   1575.00       S   
9         2013   2013  Milliers de tonnes   2824.00       S   
10        2013   2013  Milliers de tonnes   6971.00       S   
11        2013   2013                  kg    108.43      Fc   
12        2013   2013  Kcal/personne/jour    811.00      Fc   
13        2013   2013     g/personne/jour     25.99      Fc   
14        2013   2013     g/personne/jour      4.20      Fc   

   Description du Symbole  
0   Données standardisées  
1   Données standardisées  
2   Données standardisées  
3   Données standardisées  
4   Données standardisées  
5   Données standardisées  
6   Données standardisées  
7   Données standardisées  
8   Données standardisées  
9   Données standardisées  
10  Données standardisées  
11        Donnée calculée  
12        Donnée calculée  
13        Donnée calculée  
14        Donnée calculée

- Disponibilité intérieure = Production + Importations - Exportations + Variation des stocks = Nourriture + Aliments pour animaux + Semences + Pertes + Traitement + Autres Utilisations

- Nourriture / population = Disponibilité alimentaire en quantité (kg par personne par an)

## Question 3

In [17]:
# Import des données des bilans alimentaires
veg = pd.read_csv("veg.csv")
ani = pd.read_csv("ani.csv")

# Ajout de la variable origin
ani["origin"] = "animal"
veg["origin"] = "vegetal"

# On regroupe veg et ani en un unique dataframe, via une union
temp = ani.append(veg)

# Suppression de ani et veg
del ani, veg

# On renomme les colonnes de temp
temp.columns = ["xx","xx2","country_code","country",'xx3','element'
    ,'item_code','item','xx4',"year","unit","value",'xx5','xx6'
    ,'origin']

# Transformation de temp en table pivot
data = temp.pivot_table(
    index=["country_code","country","item_code","item","year","origin"],
    columns = ["element"], values=["value"], aggfunc=sum)
# On renomme les colonnes (attention l'ordre peut changer selon vos données !)
data.columns = ['domestic_supply_quantity', 'export_quantity', 'fat_supply_quantity_gcapitaday',
    'feed', 'food', 'food_supply_kcalcapitaday', 
    'food_supply_quantity_kgcapitayr', 'import_quantity', 'waste', 'other_uses',
    'processing', 'production', 'protein_supply_quantity_gcapitaday', 
    'seed', 'stock_variation']
    
data = data.reset_index()
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  food_supply_kcalcapitaday  food_supply_quantity_kgcapitayr  \
0   93.0  ...                     1024.0                           130.60   
1  137.0  ...                        0.0                             0.00   
2   96.0  ...                        0.0                             0.03   
3    1.0  ...                        1.0                             0.12   
4    4.0  ...                        2.0                             0.37   

   import_quantity  waste  other_uses  processing  production  \
0            361.0   32.0         0.0        10.0       312.0   
1              9.0   15.0        26.0         7.0       189.0   
2             82.0    7.0         NaN         NaN        21.0   
3              0.0    0.0         NaN         NaN         1.0   
4              1.0    0.0         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  
0                               30.52  30.0           -118.0  
1                                0.00  14.0              0.0  
2                                0.01   0.0              NaN  
3                                0.02   0.0              0.0  
4                                0.09   0.0              NaN  

[5 rows x 21 columns]

In [18]:
data = data.merge(pop, on=['country'])

In [19]:
data['total_kcal'] = data['food_supply_kcalcapitaday'] * 365 * data['total_population'] * 1000

In [20]:
data['total_protein_kg'] = data['protein_supply_quantity_gcapitaday'] * 1000 * 365 * data['total_population'] * 1000
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  waste  other_uses  processing  production  \
0   93.0  ...   32.0         0.0        10.0       312.0   
1  137.0  ...   15.0        26.0         7.0       189.0   
2   96.0  ...    7.0         NaN         NaN        21.0   
3    1.0  ...    0.0         NaN         NaN         1.0   
4    4.0  ...    0.0         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  \
0                               30.52  30.0           -118.0   
1                                0.00  14.0              0.0   
2                                0.01   0.0              NaN   
3                                0.02   0.0              0.0   
4                                0.09   0.0              NaN   

   total_population    total_kcal  total_protein_kg  
0              2977  1.112684e+12      3.316318e+13  
1              2977  0.000000e+00      0.000000e+00  
2              2977  0.000000e+00      1.086605e+10  
3              2977  1.086605e+09      2.173210e+10  
4              2977  2.173210e+09      9.779445e+10  

[5 rows x 24 columns]

## Question 4

In [21]:
data['kcal/kg'] = data['food_supply_kcalcapitaday'] * 365 / data['food_supply_quantity_kgcapitayr']
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  other_uses  processing  production  \
0   93.0  ...         0.0        10.0       312.0   
1  137.0  ...        26.0         7.0       189.0   
2   96.0  ...         NaN         NaN        21.0   
3    1.0  ...         NaN         NaN         1.0   
4    4.0  ...         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  \
0                               30.52  30.0           -118.0   
1                                0.00  14.0              0.0   
2                                0.01   0.0              NaN   
3                                0.02   0.0              0.0   
4                                0.09   0.0              NaN   

   total_population    total_kcal  total_protein_kg      kcal/kg  
0              2977  1.112684e+12      3.316318e+13  2861.868300  
1              2977  0.000000e+00      0.000000e+00          NaN  
2              2977  0.000000e+00      1.086605e+10     0.000000  
3              2977  1.086605e+09      2.173210e+10  3041.666667  
4              2977  2.173210e+09      9.779445e+10  1972.972973  

[5 rows x 25 columns]

In [22]:
temp = data[data['item'].str.contains("Eggs")]

In [23]:
# valeur nutritionnelle pour 100g
egg_ve = temp.replace([np.inf, -np.inf], np.nan).dropna(subset=["kcal/kg"], how="all")
egg_ve['kcal/kg'].mean()/10

136.48177721733776

- ce chiffre est inférieur en moyenne dans le monde à la valeur de 147 kcal pour un oeuf de poule entier cru donné par Wikipedia (https://fr.wikipedia.org/wiki/Œuf_(aliment))

In [24]:
data['protein_percentage'] = ((data['protein_supply_quantity_gcapitaday'] * 1000 * 365) / (data['food'] * 1000000)) * 100

In [25]:
data.head(20)

country_code  country  item_code                        item  year  \
0              1  Armenia       2511          Wheat and products  2013   
1              1  Armenia       2513         Barley and products  2013   
2              1  Armenia       2514          Maize and products  2013   
3              1  Armenia       2515            Rye and products  2013   
4              1  Armenia       2516                        Oats  2013   
5              1  Armenia       2517         Millet and products  2013   
6              1  Armenia       2518        Sorghum and products  2013   
7              1  Armenia       2520              Cereals, Other  2013   
8              1  Armenia       2531       Potatoes and products  2013   
9              1  Armenia       2532        Cassava and products  2013   
10             1  Armenia       2534                Roots, Other  2013   
11             1  Armenia       2537                  Sugar beet  2013   
12             1  Armenia       2542      Sugar (Raw Equivalent)  2013   
13             1  Armenia       2543           Sweeteners, Other  2013   
14             1  Armenia       2546                       Beans  2013   
15             1  Armenia       2547                        Peas  2013   
16             1  Armenia       2549  Pulses, Other and products  2013   
17             1  Armenia       2551           Nuts and products  2013   
18             1  Armenia       2555                   Soyabeans  2013   
19             1  Armenia       2556     Groundnuts (Shelled Eq)  2013   

     origin  domestic_supply_quantity  export_quantity  \
0   vegetal                     554.0              1.0   
1   vegetal                     198.0              0.0   
2   vegetal                     102.0              NaN   
3   vegetal                       1.0              NaN   
4   vegetal                       6.0              NaN   
5   vegetal                       0.0              NaN   
6   vegetal                       0.0              NaN   
7   vegetal                      21.0              0.0   
8   vegetal                     672.0              9.0   
9   vegetal                       0.0              NaN   
10  vegetal                       0.0              NaN   
11  vegetal                      55.0              NaN   
12  vegetal                      93.0              2.0   
13  vegetal                       5.0              0.0   
14  vegetal                       5.0              NaN   
15  vegetal                       2.0              NaN   
16  vegetal                       3.0              0.0   
17  vegetal                       6.0              1.0   
18  vegetal                       0.0              NaN   
19  vegetal                       1.0              0.0   

    fat_supply_quantity_gcapitaday   feed  ...  processing  production  \
0                             3.60   93.0  ...        10.0       312.0   
1                             0.00  137.0  ...         7.0       189.0   
2                              NaN   96.0  ...         NaN        21.0   
3                             0.00    1.0  ...         NaN         1.0   
4                             0.03    4.0  ...         NaN         5.0   
5                              NaN    0.0  ...         NaN         NaN   
6                              NaN    0.0  ...         NaN         NaN   
7                             0.04    1.0  ...         NaN        17.0   
8                             0.12  225.0  ...        78.0       661.0   
9                              NaN    0.0  ...         NaN         NaN   
10                             NaN    0.0  ...         NaN         NaN   
11                             NaN    1.0  ...        54.0        55.0   
12                             NaN    NaN  ...         NaN         8.0   
13                             NaN    NaN  ...         NaN         2.0   
14                             NaN    5.0  ...         NaN         5.0   
15                             NaN    2.0  ...

In [26]:
temp = data[data['item'].str.contains("Oats")]

In [27]:
temp = temp.replace([np.inf, -np.inf], np.nan).dropna(subset=["protein_percentage"], how="all")

In [28]:
temp = temp[temp['protein_supply_quantity_gcapitaday'] > 0]

In [29]:
temp['protein_percentage'].mean()

2.0805797229791376

- La teneur en protéine de l'avoine est en moyenne de 2,1%

## Question 5

In [30]:
temp = data.replace([np.inf, -np.inf], np.nan).dropna(subset=["kcal/kg"], how="all")
temp.pivot_table(
    index=["item_code","item","year","origin"],
    values=["kcal/kg"], aggfunc=np.mean).sort_values(by='kcal/kg', ascending=False).head(20)

kcal/kg
item_code item                     year origin              
2577      Palm Oil                 2013 vegetal  8791.217534
2571      Soyabean Oil             2013 vegetal  8625.483876
2581      Ricebran Oil             2013 vegetal  8497.969939
2574      Rape and Mustard Oil     2013 vegetal  8368.869607
2573      Sunflowerseed Oil        2013 vegetal  8305.568299
2586      Oilcrops Oil, Other      2013 vegetal  8246.215771
2575      Cottonseed Oil           2013 vegetal  8177.340910
2578      Coconut Oil              2013 vegetal  8097.752477
2576      Palmkernel Oil           2013 vegetal  7885.602997
2582      Maize Germ Oil           2013 vegetal  7827.167158
2572      Groundnut Oil            2013 vegetal  7803.918649
2580      Olive Oil                2013 vegetal  7774.263485
2737      Fats, Animals, Raw       2013 animal   7579.278844
2579      Sesameseed Oil           2013 vegetal  7482.722820
2781      Fish, Body Oil           2013 animal   7133.945055
2740      Butter, Ghee             2013 animal   6946.278887
2562      Palm kernels             2013 vegetal  5475.000000
2556      Groundnuts (Shelled Eq)  2013 vegetal  5466.324644
2561      Sesame seed              2013 vegetal  5010.932143
2633      Cocoa Beans and products 2013 vegetal  4343.034036

- Parmi les 20 aliments les plus caloriques, on peut citer toutes les huiles, le riz, le sucre, les pois et les graines de sesame

In [31]:
temp = data.replace([np.inf, -np.inf], np.nan).dropna(subset=["protein_percentage"], how="all")
temp.pivot_table(
    index=["item_code","item","year","origin"],
    values=["protein_percentage"], aggfunc=np.mean).sort_values(by='protein_percentage', ascending=False).head(20)

protein_percentage
item_code item                       year origin                     
2734      Poultry Meat               2013 animal            12.341237
2731      Bovine Meat                2013 animal            11.315275
2733      Pigmeat                    2013 animal            10.015138
2763      Pelagic Fish               2013 animal             9.496127
2549      Pulses, Other and products 2013 vegetal            8.985491
2764      Marine Fish, Other         2013 animal             8.773286
2511      Wheat and products         2013 vegetal            7.620494
2736      Offals, Edible             2013 animal             6.747168
2556      Groundnuts (Shelled Eq)    2013 vegetal            6.529281
2744      Eggs                       2013 animal             6.449287
2805      Rice (Milled Equivalent)   2013 vegetal            6.409196
2555      Soyabeans                  2013 vegetal            5.855907
2762      Demersal Fish              2013 animal             5.741089
2547      Peas                       2013 vegetal            5.626243
2546      Beans                      2013 vegetal            5.362531
2732      Mutton & Goat Meat         2013 animal             4.518097
2735      Meat, Other                2013 animal             4.448260
2765      Crustaceans                2013 animal             4.147486
2514      Maize and products         2013 vegetal            3.441945
2848      Milk - Excluding Butter    2013 animal             3.427677

- Parmi les 20 aliments les plus riches en protéines, on peut citer:
    les viandes & poissons, les lentilles, la farine, le soja, les noix

## Question 6

In [32]:
temp = data[data['origin'] == "vegetal"]

## Question 7

In [33]:
temp['total_kcal'].sum()/1000000000/2000/365

8.2898759275

- La disponibilité intérieure mondiale de produits végétaux permettrait de nourrir 8,3 milliards de personnes
(à raison d'un apport de 2000 Calories par jour et par personnes)

In [34]:
temp['total_protein_kg'].sum()/1000000000/0.05/365

6860363.9734

- La disponibilité intérieure mondiale de protéines végétales permettrait de nourrir 6860 milliards de personnes
(à raison d'un apport moyen de 50g de protéines par jour et par personne) ???

## Question 10

In [35]:
df10 = pd.read_csv("Q10.csv", usecols = ["Produit", "Valeur"])

In [36]:
(df10['Valeur'] * 1000 / pop['total_population'].sum()) * 100

0    11.730195
Name: Valeur, dtype: float64

- Le pourcentage de la population mondiale en situation de sous-nutrition est de 11.7 %

## Question 11

In [37]:
# Liste des codes produit classés par la FAO dans la catégorie Céréales
cereals = [2511, 2513, 2514, 2515, 2516, 2517, 2518, 2520, 2805]

for item_code in cereals:
    data['is_cereal'] = np.where(data['item_code'] == item_code, True, False)

# Test if values have been updated as wanted
data.query('item_code == 2805')

country_code                             country  item_code  \
87                1                             Armenia       2805   
147               2                         Afghanistan       2805   
239               3                             Albania       2805   
332               4                             Algeria       2805   
412               7                              Angola       2805   
503               8                 Antigua and Barbuda       2805   
596               9                           Argentina       2805   
691              10                           Australia       2805   
786              11                             Austria       2805   
876              12                             Bahamas       2805   
968              14                            Barbados       2805   
1062             16                          Bangladesh       2805   
1143             17                             Bermuda       2805   
1236             19    Bolivia (Plurinational State of)       2805   
1330             20                            Botswana       2805   
1426             21                              Brazil       2805   
1515             23                              Belize       2805   
1593             25                     Solomon Islands       2805   
1688             26                   Brunei Darussalam       2805   
1781             27                            Bulgaria       2805   
1866             28                             Myanmar       2805   
1960             32                            Cameroon       2805   
2056             33                              Canada       2805   
2138             35                          Cabo Verde       2805   
2223             37            Central African Republic       2805   
2318             38                           Sri Lanka       2805   
2391             39                                Chad       2805   
2484             40                               Chile       2805   
2580             41                     China, mainland       2805   
2676             44                            Colombia       2805   
...             ...                                 ...        ...   
12990           210                              Sweden       2805   
13085           211                         Switzerland       2805   
13143           213                        Turkmenistan       2805   
13236           214           China, Taiwan Province of       2805   
13332           215         United Republic of Tanzania       2805   
13427           216                            Thailand       2805   
13517           217                                Togo       2805   
13608           220                 Trinidad and Tobago       2805   
13702           221                                Oman       2805   
13794           222                             Tunisia       2805   
13887           223                              Turkey       2805   
13979           225                United Arab Emirates       2805   
14074           226                              Uganda       2805   
14169           229                      United Kingdom       2805   
14260           230                             Ukraine       2805   
14355           231            United States of America       2805   
14443           233                        Burkina Faso       2805   
14536           234                             Uruguay       2805   
14619           235                          Uzbekistan       2805   
14715           236  Venezuela (Bolivarian Republic of)       2805   
14796           237                            Viet Nam       2805   
14887           238                            Ethiopia       2805   
14963           244                               Samoa       2805   
15056           249                               Yemen       2805   
15149           251                              Zambia       2805   
15244           2

In [38]:
data.query('item_code == 2543')

country_code                             country  item_code  \
13                1                             Armenia       2543   
99                2                         Afghanistan       2543   
164               3                             Albania       2543   
257               4                             Algeria       2543   
347               7                              Angola       2543   
430               8                 Antigua and Barbuda       2543   
520               9                           Argentina       2543   
615              10                           Australia       2543   
710              11                             Austria       2543   
802              12                             Bahamas       2543   
894              14                            Barbados       2543   
986              16                          Bangladesh       2543   
1075             17                             Bermuda       2543   
1160             19    Bolivia (Plurinational State of)       2543   
1254             20                            Botswana       2543   
1349             21                              Brazil       2543   
1442             23                              Belize       2543   
1529             25                     Solomon Islands       2543   
1612             26                   Brunei Darussalam       2543   
1705             27                            Bulgaria       2543   
1798             28                             Myanmar       2543   
1884             32                            Cameroon       2543   
1979             33                              Canada       2543   
2071             35                          Cabo Verde       2543   
2155             37            Central African Republic       2543   
2241             38                           Sri Lanka       2543   
2334             39                                Chad       2543   
2408             40                               Chile       2543   
2504             41                     China, mainland       2543   
2600             44                            Colombia       2543   
...             ...                                 ...        ...   
12914           210                              Sweden       2543   
13009           211                         Switzerland       2543   
13098           213                        Turkmenistan       2543   
13161           214           China, Taiwan Province of       2543   
13256           215         United Republic of Tanzania       2543   
13351           216                            Thailand       2543   
13443           217                                Togo       2543   
13535           220                 Trinidad and Tobago       2543   
13627           221                                Oman       2543   
13719           222                             Tunisia       2543   
13812           223                              Turkey       2543   
13904           225                United Arab Emirates       2543   
13998           226                              Uganda       2543   
14093           229                      United Kingdom       2543   
14186           230                             Ukraine       2543   
14279           231            United States of America       2543   
14372           233                        Burkina Faso       2543   
14461           234                             Uruguay       2543   
14550           235                          Uzbekistan       2543   
14639           236  Venezuela (Bolivarian Republic of)       2543   
14728           237                            Viet Nam       2543   
14813           238                            Ethiopia       2543   
14903           244                               Samoa       2543   
14981           249                               Yemen       2543   
15074           251                              Zambia       2543   
15168           2

## Question 14

- En thaïlande, 6,1 millions de personnes sont sous alimentées en 2013

In [39]:
pop.query('country == "Thailand"')

total_population
country_code country  item_code item       year                  
216          Thailand 2501      Population 2013             67011

In [40]:
(6100000/(67011000)) * 100

9.102983092328126

- Soit 9.1 % de la population de ce pays.

In [41]:
temp = data[data['item'].str.contains("Cassava")]

In [42]:
temp = temp.query('country == "Thailand"')

In [43]:
temp['export_quantity'] / temp['production'] * 100

13344    83.41273
dtype: float64

- La proportion de manioc exportée de la Thaïlande est de 83.4 %